custom/sensors/PDP.java

```diff
@@ -56,36 +56,54 @@ public PDP() {
 	private void readStatus1() {
 		byte[] rawArray = status1.read();
 		if (rawArray != null) {
-			cachedChannelCurrents[0] = (((rawArray[0] & 0xFF) << 2) | rawArray[1] & 0xC0) * 0.125;
-			cachedChannelCurrents[1] = (((rawArray[1] & 0x3F) << 4) | rawArray[2] & 0xF0) * 0.125;
-			cachedChannelCurrents[2] = (((rawArray[2] & 0x0F) << 6) | rawArray[3] & 0x3F) * 0.125;
-			cachedChannelCurrents[3] = (((rawArray[3] & 0xC0) << 8) | rawArray[4] & 0xFF) * 0.125;
-			cachedChannelCurrents[4] = (((rawArray[5] & 0xFF) << 2) | rawArray[6] & 0xC0) * 0.125;
-			cachedChannelCurrents[5] = (((rawArray[6] & 0x3F) << 4) | rawArray[7] & 0xF0) * 0.125;
+			double[] tempCurrents = new double[6];
+			tempCurrents[0] = ((rawArray[0] & 0xFF) << 2 | ((rawArray[1] & 0xC0) >> 6)) * 0.125;
+			tempCurrents[1] = (((rawArray[1] & 0x3F) << 4) | ((rawArray[2] & 0xF0) >> 4)) * 0.125;
+			tempCurrents[2] = (((rawArray[2] & 0x0F) << 6) | ((rawArray[3] & 0x3F) >> 2)) * 0.125;
+			tempCurrents[3] = (((rawArray[3] & 0xC0) << 8) | ((rawArray[4] & 0xFF))) * 0.125;
+			tempCurrents[4] = (((rawArray[5] & 0xFF) << 2) | ((rawArray[6] & 0xC0) >> 6)) * 0.125;
+			tempCurrents[5] = (((rawArray[6] & 0x3F) << 4) | ((rawArray[7] & 0xF0) >> 4)) * 0.125;
+			for (int i = 0; i < 6; i++) {
+				if (tempCurrents[i] < 128) { // deals with occasional issue with PDP reporting 1000+ amps (this is not a bug in this code, it was observed in PowerDistributionPanel as well
```

Comment:
Perhaps now that there is actual logic here the three readStatus methods should be generalized into one?